test/e2e/plain-v0_provisioner_test.go

```diff
@@ -192,3 +194,188 @@ var _ = Describe("plain-v0 provisioner", func() {
+					return false
+				}
+				existing := meta.FindStatusCondition(bi.Status.Conditions, "Installed")
+				if existing == nil {
+					return false
+				}
+				expected := metav1.Condition{
+					Type:    "Installed",
+					Status:  metav1.ConditionStatus(corev1.ConditionTrue),
+					Reason:  "InstallationSucceeded",
+					Message: "",
+				}
+				return conditionsSemanticallyEqual(expected, *existing)
+			}).Should(BeTrue())
+
+			By("eventually reseting a bundle lookup failure when the targeted bundle has been deleted")
+			Eventually(func() error {
+				return c.Delete(ctx, bundle)
+			}).Should(Succeed())
+
+			Eventually(func() bool {
+				if err := c.Get(ctx, client.ObjectKeyFromObject(bi), bi); err != nil {
+					return false
+				}
+				if bi.Status.InstalledBundleName != "" {
+					return false
+				}
+				existing := meta.FindStatusCondition(bi.Status.Conditions, "Installed")
+				if existing == nil {
+					return false
+				}
+				expected := metav1.Condition{
+					Type:    "Installed",
+					Status:  metav1.ConditionStatus(corev1.ConditionFalse),
+					Reason:  "BundleLookupFailed",
+					Message: fmt.Sprintf(`Bundle.core.rukpak.io "%s" not found`, bundle.GetName()),
+				}
+				return conditionsSemanticallyEqual(expected, *existing)
+			}).Should(BeTrue())
+		})
+	})
+
+	When("a BundleInstance targets an invalid Bundle", func() {
+		var (
+			bundle *v1alpha1.Bundle
+			bi     *v1alpha1.BundleInstance
+		)
+		BeforeEach(func() {
+			By("creating the testing Bundle resource")
+			bundle = &v1alpha1.Bundle{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "olm-apis",
+				},
+				Spec: v1alpha1.BundleSpec{
+					ProvisionerClassName: plainProvisionerID,
+					Image:                "quay.io/tflannag/olm-plain-bundle:olm-api-v0.20.0",
+				},
+			}
+			err := c.Create(ctx, bundle)
+			Expect(err).To(BeNil())
+
+			bi = &v1alpha1.BundleInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "olm-apis",
+				},
+				Spec: v1alpha1.BundleInstanceSpec{
+					ProvisionerClassName: plainProvisionerID,
+					BundleName:           bundle.GetName(),
+				},
+			}
+			err = c.Create(ctx, bi)
+			Expect(err).To(BeNil())
+		})
+		AfterEach(func() {
+			By("deleting the testing Bundle resource")
+			Eventually(func() error {
+				return client.IgnoreNotFound(c.Delete(ctx, bundle))
+			}).Should(Succeed())
+
+			By("deleting the testing BundleInstance resource")
+			Eventually(func() error {
+				return client.IgnoreNotFound(c.Delete(ctx, bi))
+			}).Should(Succeed())
+		})
+
+		It("should project a failed installation state", func() {
+			Eventually(func() bool {
+				if err := c.Get(ctx, client.ObjectKeyFromObject(bi), bi); err != nil {
+					return false
+				}
+				if bi.Status.InstalledBundleName != "" {
+					return false
+				}
+				existing := meta.FindStatusCondition(bi.Status.Conditions, "Installed")
+				if existing == nil {
+					return false
+				}
+				// TODO(tflannag): This kind of comparison feels brittle?
+				expectedMessage := `
```

Comment:
this is the error we get when attempting to install a CR for a CRD that is not on the cluster? Maybe we should have custom error handling for this type of error as it seems pretty common

Reply:
There's a bunch of potential cleanup needed in the BundleInstance controller around hardcoding condition types/reasons/phases/etc. so I didn't want to float this PR with making those kind of changes all at once. I've largely agree that this kind of installation failure should warrant it's own error type though. Maybe I should just add a TODO for now and we can revisit next week?